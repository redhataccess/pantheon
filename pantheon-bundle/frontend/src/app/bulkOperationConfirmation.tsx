import React from 'react';
import { Modal, ModalVariant, Button, Title, TitleSizes, AlertActionCloseButton, Alert, AlertActionLink, Progress, ProgressVariant, ProgressSize, List, ListItem, ProgressMeasureLocation } from '@patternfly/react-core';
import WarningTriangleIcon from '@patternfly/react-icons/dist/js/icons/warning-triangle-icon';
import "@app/app.css";

export interface IBulkOperationProps {
  header: string
  subheading: string
  updateSucceeded: string
  updateFailed: string
  updateIgnored: string
  footer: string
  progressSuccessValue: number
  progressFailureValue: number
  progressWarningValue: number
  onShowBulkEditConfirmation: (showBulkEditConfirmation) => any
  onMetadataEditError: (metadataEditError) => any
}

class BulkOperationConfirmation extends React.Component<IBulkOperationProps, any>{
  constructor(props) {
    super(props);
    this.state = {
      isModalOpen: false,
    };

  }

  render() {
    const { isModalOpen } = this.state;

    const header = (
      <React.Fragment>
        <Title id="bulk-operation-more-details" headingLevel="h1" size={TitleSizes['2xl']}>
          {this.props.header}
        </Title>
        <p className="pf-u-pt-sm">{this.props.subheading}</p>
      </React.Fragment>
    );

    const footer = (
      <Title headingLevel="h4" size={TitleSizes.md}>
        <span className="pf-u-pl-sm">{this.props.footer}</span>
      </Title>
    );

    return (
      <React.Fragment>
          <Alert
            variant="info"
            title="Bulk Edit"
            actionClose={<AlertActionCloseButton data-testid="hide-alert-button" onClose={this.hideAlert} />}
            actionLinks={
              <React.Fragment>
                <AlertActionLink data-testid="view-details-link" onClick={this.handleModalToggle}>View details</AlertActionLink>
              </React.Fragment>
            }
          >
            <div><Progress value={this.props.progressSuccessValue} title="Update Succeeded" variant={ProgressVariant.success} size={ProgressSize.sm} /></div>
            <div><Progress value={this.props.progressFailureValue} title="Update failed" variant={ProgressVariant.danger} size={ProgressSize.sm} /></div>
            <div><Progress value={this.props.progressWarningValue} title="No draft version found. No action taken" variant={ProgressVariant.warning} size={ProgressSize.sm} /></div>
          </Alert>
        <Modal
          variant={ModalVariant.large}
          isOpen={isModalOpen}
          header={header}
          aria-label="confirmation dialog"
          aria-labelledby="confirmation-header-label"
          aria-describedby="confirmation-header-description"
          onClose={this.handleModalToggle}
          footer={footer}
        >
          <strong>Update Succeeded:</strong>
          <br />
          <span id="update-succeeded">
            <List aria-label="succeeded">
              {this.props.updateSucceeded.length > 0 &&
                this.props.updateSucceeded.split(",").map((data) => (
                  data.length > 0 &&
                  <ListItem>{data}</ListItem>
                ))}
            </List>
          </span>
          <br />
          <br />
          <strong>Update Ignored:</strong>
          <br />
          <span id="update-ignored">
            <List aria-label="ignored">
              {this.props.updateIgnored.length > 0 &&
                this.props.updateIgnored.split(",").map((data) => (
                  data.length > 0 &&
                  <ListItem>{data}</ListItem>
                ))}
            </List>
          </span>
          <br />
          <br />
          <strong>Update Failed:</strong>
          <br />
          <span id="update-failed">
            <List aria-label="failed">
              {this.props.updateFailed.length > 0 &&
                this.props.updateFailed.split(",").map((data) => (
                  data.length > 0 &&
                  <ListItem>{data}</ListItem>
                ))}
            </List>
          </span>
          <br />
          <br />

        </Modal>
      </React.Fragment>
    );
  }

  private handleModalToggle = () => {
    this.setState(({ isModalOpen }) => ({
      isModalOpen: !isModalOpen
    }));
  };

  private hideAlert = () => {
    this.props.onShowBulkEditConfirmation(false)
    this.props.onMetadataEditError("")
    //TODO: refresh documentsSelected
    // this.SearchResults.current.doSearch()
  }
}

export { BulkOperationConfirmation }