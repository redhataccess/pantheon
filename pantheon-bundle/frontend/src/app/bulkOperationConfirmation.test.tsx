import React from "react"
import { BulkOperationConfirmation } from "./bulkOperationConfirmation";

import { mount, shallow } from "enzyme"
import { Alert, Modal, Progress } from "@patternfly/react-core"
import { render, fireEvent } from '@testing-library/react'

const anymatch = require("anymatch")

const props = {
    isEditMetadata: true,
    header: "Bulk Edit",
    subheading: "Summary",
    updateSucceeded: "/repositories/s/a,/repositories/s/b,/repositories/s/c",
    updateFailed: "/repositories/f/a",
    updateIgnored: "/repositorires/i/a,/repositorires/i/b",
    footer: "",
    progressSuccessValue: 50,
    progressFailureValue: 10,
    progressWarningValue: 40,
    onShowBulkEditConfirmation: (showBulkEditConfirmation) => anymatch,
    onMetadataEditError: (metadataEditError) => anymatch,
    updateIsEditMetadata: (isEditMetadata) => anymatch,
    onProgressSuccessValue: (progressSuccessValue) => anymatch,
    onProgressFailureValue: (progressFailureValue) => anymatch,
    onProgressWarningValue: (progressWarningValue) => anymatch,
    onUpdateSucceeded: (updateSucceeded) => anymatch,
    onUpdateIgnored: (updateIgnored) => anymatch,
    onUpdateFailed: (updateFailed) => anymatch
}

describe("BulkOperationConfirmation tests", () => {
    test("should render BulkOperationConfirmation component", () => {
        const view = shallow(<BulkOperationConfirmation {...props} />)
        expect(view).toMatchSnapshot()
    })

    it("should render an Alert component", () => {
        const wrapper = mount(<BulkOperationConfirmation {...props} />)
        const alert = wrapper.find(Alert)
        expect(alert.exists()).toBe(true)
    })


    it("should render a Modal component", () => {
        const wrapper = mount(<BulkOperationConfirmation {...props} />)
        const modal = wrapper.find(Modal)
        expect(modal.exists()).toBe(true)
    })

    it("should render a Progress component", () => {
        const wrapper = mount(<BulkOperationConfirmation {...props} />)
        const progress = wrapper.find(Progress)
        expect(progress.exists()).toBe(true)
    })

    it("test hideAlert function", () => {
        // render component  
        const { getByTestId } = render(<BulkOperationConfirmation {...props} />)

        // click "Hide Alert" button
        fireEvent.click(getByTestId("hide-alert-button"))

        // verify the alert is gone
        expect(getByTestId("hide-alert-button")).toBeFalsy
    })

    it("test handleModalToggle function", () => {
        // render component  
        const { getByTestId, queryAllByText } = render(<BulkOperationConfirmation {...props} />)

        // click "view details" link
        fireEvent.click(getByTestId("view-details-link"))

        // verify
        expect(queryAllByText("Bulk Edit")).toBeTruthy
    })

})
