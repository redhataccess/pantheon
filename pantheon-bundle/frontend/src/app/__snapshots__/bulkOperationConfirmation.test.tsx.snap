// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BulkOperationConfirmation tests should render BulkOperationConfirmation component 1`] = `
<Fragment>
  <Alert
    actionClose={
      <AlertActionCloseButton
        data-testid="hide-alert-button"
        onClose={[Function]}
      />
    }
    actionLinks={
      <React.Fragment>
        <AlertActionLink
          data-testid="view-details-link"
          onClick={[Function]}
        >
          View details
        </AlertActionLink>
      </React.Fragment>
    }
    title="Bulk Edit"
    variant="info"
  >
    <div>
      <Progress
        className=""
        id=""
        isTitleTruncated={false}
        label={null}
        max={100}
        measureLocation="top"
        min={0}
        size="sm"
        title="Update Succeeded"
        tooltipPosition="top"
        value={50}
        valueText={null}
        variant="success"
      />
    </div>
    <div>
      <Progress
        className=""
        id=""
        isTitleTruncated={false}
        label={null}
        max={100}
        measureLocation="top"
        min={0}
        size="sm"
        title="Update failed"
        tooltipPosition="top"
        value={10}
        valueText={null}
        variant="danger"
      />
    </div>
    <div>
      <Progress
        className=""
        id=""
        isTitleTruncated={false}
        label={null}
        max={100}
        measureLocation="top"
        min={0}
        size="sm"
        title="No draft version found. No action taken"
        tooltipPosition="top"
        value={40}
        valueText={null}
        variant="warning"
      />
    </div>
  </Alert>
  <Modal
    actions={Array []}
    appendTo={[Function]}
    aria-describedby="confirmation-header-description"
    aria-label="confirmation dialog"
    aria-labelledby="confirmation-header-label"
    className=""
    footer={
      <Title
        headingLevel="h4"
        size="md"
      >
        <span
          className="pf-u-pl-sm"
        >
          
        </span>
      </Title>
    }
    hasNoBodyWrapper={false}
    header={
      <React.Fragment>
        <Title
          headingLevel="h1"
          id="bulk-operation-more-details"
          size="2xl"
        >
          Bulk Edit
        </Title>
        <p
          className="pf-u-pt-sm"
        >
          Summary
        </p>
      </React.Fragment>
    }
    isOpen={false}
    onClose={[Function]}
    ouiaSafe={true}
    showClose={true}
    title=""
    titleIconVariant={null}
    titleLabel=""
    variant="large"
  >
    <strong>
      Update Succeeded:
    </strong>
    <br />
    <span
      id="update-succeeded"
    >
      <List
        aria-label="succeeded"
      >
        <ListItem>
          /repositories/s/a
        </ListItem>
        <ListItem>
          /repositories/s/b
        </ListItem>
        <ListItem>
          /repositories/s/c
        </ListItem>
      </List>
    </span>
    <br />
    <br />
    <strong>
      Update Ignored:
    </strong>
    <br />
    <span
      id="update-ignored"
    >
      <List
        aria-label="ignored"
      >
        <ListItem>
          /repositorires/i/a
        </ListItem>
        <ListItem>
          /repositorires/i/b
        </ListItem>
      </List>
    </span>
    <br />
    <br />
    <strong>
      Update Failed:
    </strong>
    <br />
    <span
      id="update-failed"
    >
      <List
        aria-label="failed"
      >
        <ListItem>
          /repositories/f/a
        </ListItem>
      </List>
    </span>
    <br />
    <br />
  </Modal>
</Fragment>
`;
